docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,247 @@
+
+### Templating
+
+Chosen option: "Option B - Use React server-side to generate static pages", because:
+
+- The team has experience with (and likes) React
+- We're already using it in other parts of the FxA ecosystem
+- React has been tried and tested to work well with Storybook
+- React can be integrated with mjml to generate responsive emails and there is already a great [library](https://github.com/wix-incubator/mjml-react) which can help us with that purpose.
+- Components across the templates can be reused if written in React
+
+A note about tests: Since we're going with the React-based solution for our templating stack, it may make sense to use Jest to test the templates. However, we don't want a ton of time spent in test land. At the very least, we can take what we currently have and transfer/copy it, and look them over to see if anything can be improved without a high level of effort.
+
+### CSS Styles
+
+Chosen option: [todo], because:
+
+- [todo]
+
+### Email Previewing
+
+Chosen option: "Option B - Use Storybook", because:
+
+- It is easy to setup and configure.
+- We have a Storybook deployment already in place and has worked great so far for a few other FxA services so hooking up the client for `fxa-auth-server` to view the modernized components might result in consistency.
+- Easy to test and perform QA. The design and product teams don't have to touch the codebase to test the various states of UI.
+- Support for CSS (whether it's plain old CSS, TailwindCSS or something else)
+
+### Other
+
+**Plaintext files**: [todo]
+
+**Localization**: We will upgrade the localization tool from GetText to Fluent since it's preferred by the l10n team and other FxA front-ends are using it.
+
+**Integration & QA**: [todo]
+We also intend on making a plan on how we're going to build out/hook refactored templates up with feature flags and mitigate risk. The `fxa-auth-server` already has some basic support for feature flagging, but this ADR aims to identify how we want to use these - do we want 3 different feature flags (one for each milestone of templates), do we want to flag individual templates, or should we have a single flag?
+
+### Pros and Cons of using MJML as part of our templating stack
+
+As part of modernizing the templates, we aim to replace the framework in which our templates are currently built in so as to include more logic in them while development. Along with the framework, we've decided to use [MJML](https://mjml.io/) as our potential markup language, designed to reduce the pain of coding a responsive email. However, there needs to be more discussion around it so the following list of pros and cons could be pondered upon to decide whether or not we should use MJML as part of our templating stack.
+
+- Pros
+  - MJML is a component-based library having its own custom components defined which translates to responsive HTML after compilation
+  - MJML can help us create responsive emails using mobile-first approach
```

Comment:
I'm not 100% sure how much we gain from responsive emails given the current design of them. Maybe this could eventually lead to a redesign.